src/error.rs

```diff
@@ -15,6 +59,22 @@ pub type Result<T> = ::std::result::Result<T, Error>;
 
 impl Error {
     /// Returns a possible URL related to this error.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # extern crate reqwest;
+    /// # fn main() {
+    /// let response = reqwest::get("");
+    /// match response {
+    ///   Err(e) => match e.url() {
+    ///     None => println!("No Url given"),
+    ///     Some(url) => println!("Bad Url given: {}", url),
```

Comment:
This error probably won't have a URL associated with it, since it wasn't able to parse one. The `url()` method will have a `Url` if reqwest was able to parse the error and start an HTTP request. Situations this would have a `Url` and be useful for:

- The redirect policy notices a loop. You can check the `url()` for the last request made that determined there was a loop.
- There was a network error while making the request. Again, also useful if after a redirect, the server aborted the socket early, while still emitting headers. You could see the `io::Error` occurred from which server.